common/src/main/java/com/wynntils/core/features/overlays/annotations/OverlayInfo.java

```diff
@@ -20,6 +20,8 @@
 
     RenderState renderAt() default RenderState.Post;
 
+    boolean cancelRender() default false;
```

Comment:
You set this to true to override the vanilla rendering? It's not super clear. 

What about having the render state being a tri-state instead, Pre, Post and Replace? The latter would be called at Pre but also cancel the event. And you could skip the check in the init that we don't try a cancelRender for Post.